src/main/java/httpserver/ServerSocketException.java

```diff
@@ -0,0 +1,7 @@
+package httpserver;
+
+public class ServerSocketException extends RuntimeException {
+    ServerSocketException(Exception cause) {
+        super(cause);
```

Comment:
Rather than have your exception handler check for `instanceof`, could your custom exceptions be altered so that they contain all the necessary information? That way, your handler doesn't need to change when you add new Exception types. Would it make sense to devise an interface so that you could do something like `myException.printableMessage()`?

Reply:
Thanks, Ashley. I'll look into this